### PR TITLE
[#566] Add user to shelf (connects #566)

### PR DESCRIPTION
**Pull request by Rupesh87**

### CODE REVIEW

File: src/app/users/users.component.html [hunk 1]
```diff
@@ -83,6 +84,8 @@ <h1 i18n>All Users</h1>
           <button mat-icon-button *ngIf="element.roles.length > 0" (click)="removeRole(element)">
             <mat-icon>lock_open</mat-icon>
           </button>
+          <button mat-raised-button color="primary"  *ngIf="!element.myTeamInfo" (click)="addTeams([{'_id': element._id}])">Add Team</button>
+          <button mat-raised-button color="primary"  *ngIf="element.myTeamInfo" (click)="removeTeam(element._id)">Remove Team</button>
```
Comment (nirojdyola) [inline src/app/users/users.component.html]: we can use it as in else part using ng-container. Could you please check it once ?

File: src/app/users/users.component.ts [hunk 2]
```diff
@@ -85,15 +87,32 @@ export class UsersComponent implements OnInit, AfterViewInit {
...
+      this.allUsers.data = data[0].rows.reduce((users: any[], user: any) => {
         if (user.id !== '_design/_auth') {
           if (user.doc._attachments) {
             user.doc.imageSrc = this.urlPrefix + 'org.couchdb.user:' + user.doc.name + '/' + Object.keys(user.doc._attachments)[0];
           }
-          users.push({ ...user.doc });
+          const myTeamIndex = data[1].docs[0].myTeamIds ? data[1].docs[0].myTeamIds.findIndex(myTeamId => {
+            return user.id === myTeamId;
+          }) : -1;
+          if (myTeamIndex > -1) {
```
Comment (nirojdyola) [inline src/app/users/users.component.ts]: can minimize the code using ternary operator.

File: src/app/users/users.component.ts [hunk 3]
```diff
@@ -85,15 +87,28 @@ export class UsersComponent implements OnInit, AfterViewInit {
...
         if (user.id !== '_design/_auth') {
           if (user.doc._attachments) {
             user.doc.imageSrc = this.urlPrefix + 'org.couchdb.user:' + user.doc.name + '/' + Object.keys(user.doc._attachments)[0];
           }
-          users.push({ ...user.doc });
+          const myTeamIndex = data[1].docs[0].myTeamIds ? data[1].docs[0].myTeamIds.findIndex(myTeamId => {
+            return user.id === myTeamId;
+          }) : -1;
+          myTeamIndex > -1 ? users.push({ ...user.doc, myTeamInfo: true }) : users.push({ ...user.doc, myTeamInfo: false });
         } else if (user.id !== '_design/_auth' && user.doc.isUserAdmin === true) {
```
Comment (Rupesh87) [inline src/app/users/users.component.ts]: this condition will never run

File: src/app/users/users.component.ts [hunk 4]
```diff
@@ -179,6 +194,55 @@ export class UsersComponent implements OnInit, AfterViewInit {
...
   }
 
+  dedupeShelfReduce(ids, id) {
+    if (ids.indexOf(id) > -1) {
+      return ids;
+    }
+    return ids.concat(id);
+  }
+
+  removeAdminId(ids, id) {
```
Comment (paulbert) [inline src/app/users/users.component.ts]: We don't need to remove an admin from this.

File: src/app/users/users.component.ts (hunk 3 above)
Comment (paulbert) [reply src/app/users/users.component.ts]: Made issue #602 to cover this

**Pull request merged**